liquid-staking/src/funds/delegate_vote.rs

```diff
@@ -0,0 +1,66 @@
+multiversx_sc::imports!();
+
+use crate::{
+    basics::errors::{
+        ERROR_BAD_PAYMENT_AMOUNT, ERROR_BAD_PAYMENT_TOKEN, ERROR_LS_TOKEN_NOT_ISSUED,
+    },
+    contexts::base::StorageCache,
+    liquidity_pool,
+    proxies::vote_proxy::{self, VoteType},
+    setup,
+};
+
+#[multiversx_sc::module]
+pub trait DelegateVoteModule:
+    setup::config::ConfigModule
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+    + liquidity_pool::LiquidityPoolModule
+    + setup::vote::VoteModule
+{
+    #[payable("*")]
+    #[endpoint(delegateVote)]
+    fn delegate_vote(&self, proposal: usize, vote_type: VoteType) {
+        self.blockchain().check_caller_is_user_account();
+        let storage_cache = StorageCache::new(self);
+        let caller = self.blockchain().get_caller();
+        let payment = self.call_value().single_esdt();
+
+        require!(
+            storage_cache.ls_token_id.is_valid_esdt_identifier(),
+            ERROR_LS_TOKEN_NOT_ISSUED
+        );
+        require!(
+            payment.token_identifier == storage_cache.ls_token_id,
+            ERROR_BAD_PAYMENT_TOKEN
+        );
+        require!(payment.amount > 0, ERROR_BAD_PAYMENT_AMOUNT);
+
+        let balance_to_vote = self.get_egld_amount(&payment.amount, &storage_cache);
+
+        self.call_delegate_vote(proposal, vote_type, &caller, balance_to_vote);
+
+        // we use the ls token only to access amount staked, afterwards we send it back to the caller
+        self.send().direct_esdt(
```

Comment:
This doesn't feel right. We need to know how the voting SC would look like, but I don't really think it can cover all the possible exploit cases.
What happens if after I call delegateVote, I make a new wallet (still mine) and call delegateVote once again from that wallet. I can game the logic on loop.